src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexRetriever.java

```
@@ -0,0 +1,187 @@
+    public static final String KEY = "GLOBAL_STOCK_INDEX";
+
+    @Value("${morningstar.username}")
+    private String ftpUsername;
+
+    @Value("${morningstar.password}")
+    private String ftpPassword;
+
+    @Value("${morningstar.host}")
+    private String ftpHost;
+
+    @Value("${morningstar.port}")
+    private int ftpPort;
+
+    private static final String PATH = "/Daily/DMRI/XI_MSTAR/";
+    private static final String secID = "F00000VN9N";
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public List<FundValue> retrieveValuesForRange(LocalDate startDate, LocalDate endDate) {
+        ArrayList<DailyRecord> records = new ArrayList<>();
+
+        FtpClient ftpClient = new FtpClient(ftpHost, ftpUsername, ftpPassword, ftpPort);
+        try {
+            ftpClient.open();
+            Collection<String> fileNames = ftpClient.listFiles(GlobalStockIndexRetriever.PATH);
+            for(LocalDate date = startDate; date.isBefore(endDate) || date.isEqual(endDate); date = date.plusDays(1))
+            {
+                String dateString = date.format(DateTimeFormatter.ofPattern("yyyyMMdd"));
+                String fileName = fileNames.stream()
+                    .filter(string -> string.endsWith(dateString + ".zip"))
+                    .findAny()
+                    .orElse(null);
+
+                if(fileName == null) {
+                    continue;
+                }
+
+                try{
+                    InputStream fileStream = ftpClient.downloadFileStream(GlobalStockIndexRetriever.PATH + fileName);
+                    Optional<DailyRecord> optionalRecord = findInZip(fileStream, GlobalStockIndexRetriever.secID);
+                    optionalRecord.ifPresent(dailyRecord -> pushDailyRecord(records, dailyRecord));
+                    fileStream.close();
+                    ftpClient.completePendingCommand();
+                } catch(IOException ignored) {}
```

Never ignore exceptions.